docs/src/main/docs/config/07_extensions.adoc

```diff
@@ -305,32 +305,53 @@ types. (See link:{javadoc-base-url-api}/ConfigMappers.html[`ConfigMappers`].)
 To handle mappings to other types the application can register
 custom mappers with the config system by implementing the 
 link:{javadoc-base-url-api}/spi/ConfigMapperProvider.html[`ConfigMapperProvider`]
- SPI. Such providers return a map, with entries in which:
+ SPI.
+
+Such providers return a map, with entries in which:
 
 * the key is the Java type (a `Class` object) the mapper produces, and
 * the value is a `ConfigMapper` that converts the config in-memory
 data structure into the type in the key.
 
+The provider may also implement other methods for finer tuned conversion mechanisms:
+
+* `genericTypeMappers()` returns a map with entries for specific `GenericType` conversions,
+    e.g. when the provider supports only mapping for `GenericType<Map<String, Integer>>`
+* `mapper(Class)` returns a conversion function (optional) that converts a config node
+    to the typed instance (if supported by this provider)
+* `mapper(GenericType)` returns a conversion function (optional) that coverts a config node
+    to the GenericType (if supported by this provider) - e.g. in case this provider supporst
+    any Map<String, ?> type, such as `Map<String, Integer>` and `Map<String, Double>`
+
+The config conversion system works as follows:
+
+For `Config.as(Class)`:
+
+1. check if a conversion function exists for the class requested (from method `mappers()`)
+2. check if a conversion function is provided by any `ConfigMapperProvider` with method `mapper(Class)`
+3. check if a conversion function exists for a generic type for the class requested (from method `genericTypeMappers`)
+4. check if a conversion function is provided by any `ConfigMapperProvider` with method `mapper(GenericType)` for
+    a generic type for the class requested
+
```

Comment:
Minor comment: Change "check if..." to "Check whether..." and end each step with a period.